Sources/Datadog/RUM/RUMVitals/VitalInfo.swift

```diff
@@ -6,16 +6,53 @@
 
 import Foundation
 
+internal typealias VitalPublisher = ValuePublisher<VitalInfo>
+
 internal struct VitalInfo {
     /// Number of sample for this info
-    internal let sampleCount: Int
-
+    var sampleCount: Int { Int(sampleCountDouble) }
+    /// To avoid int->double conversion too often
+    private var sampleCountDouble = 0.0
     /// Minimum value across all samples
-    internal let minValue: Double
-
+    private(set) var minValue: Double?
     /// Maximum value across all samples
-    internal let maxValue: Double
-
+    private(set) var maxValue: Double?
     /// Average value across all samples
-    internal let meanValue: Double
+    private(set) var meanValue: Double?
+    /// Diff between max and min values
+    var greatestDiff: Double? {
+        if let someMax = maxValue, let someMin = minValue {
+            return someMax - someMin
+        }
+        return nil
+    }
+
+    mutating func addSample(_ sample: Double) {
+        // Assuming M(n) is the mean value of the first n samples
+        // M(n) = ∑ sample(n) / n
+        // n⨉M(n) = ∑ sample(n)
+        // M(n+1) = ∑ sample(n+1) / (n+1)
+        //        = [ sample(n+1) + ∑ sample(n) ] / (n+1)
+        //        = (sample(n+1) + n⨉M(n)) / (n+1)
+        meanValue = (sample + (sampleCountDouble * (meanValue ?? 0.0))) / (sampleCountDouble + 1.0)
+        // swiftlint:disable force_unwrapping
+        minValue = minValue == nil ? sample : min(minValue!, sample)
+        maxValue = maxValue == nil ? sample : max(maxValue!, sample)
```

Comment:
Any reason for force unwrapping instead of `if let` 💭 ?

Reply:
just for the sake of brevity, it becomes longer otherwise
```swift
if let someMin = minValue {
  minValue = min(someMin, sample)
} else {
  minValue = sample
}
// x2, same for max
```

can you think of any potential problems with force unwrap here?

Reply:
Force unwrapping is generally considered a bad practise hence we control it with linter (explicitly disabled here). It won't cause any issue in this case 👌.